widget/hexwidget/hexwidget.go

```diff
@@ -0,0 +1,240 @@
+// Package hexwidget implements a 7-segment style hexadecimal display
+package hexwidget
+
+import (
+	"image"
+	"image/color"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/canvas"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+// segmentLookupTable is used by h.Set() - the i-th index into this table
+// represents the raw value that should be sent to updateSegments to show
+// the value i.
+var segmentLookupTable []uint8 = []uint8{
+	1 << 6,
+	(1<<0 | (1 << 1) | (1 << 2) | (1 << 3) | (1 << 6) | (1 << 7)),
+	(1<<2 | (1 << 5)),
+	(1<<4 | (1 << 5)),
+	(1<<0 | (1 << 3) | (1 << 4)),
+	(1<<1 | (1 << 4)),
+	(1 << 1),
+	(1<<3 | (1 << 4) | (1 << 5) | (1 << 6)),
+	0,
+	(1<<3 | (1 << 4)),
+	(1 << 3),
+	(1<<0 | (1 << 1)),
+	(1<<0 | (1 << 1) | (1 << 2) | (1 << 5)),
+	(1<<0 | (1 << 5)),
+	(1<<1 | (1 << 2)),
+	(1<<1 | (1 << 2) | (1 << 3)),
+}
+
+// size in pixels of the hex widget
+const defaultHexHeight float32 = 75.0
+const defaultHexWidth float32 = defaultHexHeight * (7.5 / 14.0)
+
+// slant angle
+const defaultHexOffset float32 = 0.1 * defaultHexWidth
+
+var defaultHexOnColor color.RGBA = color.RGBA{200, 25, 25, 255}
+var defaultHexOffColor color.RGBA = color.RGBA{25, 15, 15, 64}
+
+type hexRenderer struct {
+	hex            *HexWidget
+	segmentObjects []fyne.CanvasObject
+}
+
+func (h *hexRenderer) MinSize() fyne.Size {
+	return fyne.NewSize(
+		float32(h.hex.hexWidth)+theme.Padding()*2+h.hex.hexOffset,
+		float32(h.hex.hexHeight)+theme.Padding()*2,
+	)
+}
+
+func (h *hexRenderer) Layout(size fyne.Size) {
+}
+
+func (h *hexRenderer) ApplyTheme() {
+}
+
+func (h *hexRenderer) BackgroundColor() color.Color {
+	return theme.BackgroundColor()
+}
+
+func (h *hexRenderer) Refresh() {
+	hexSegmentWidth := 0.2 * h.hex.hexWidth
+	hexSegmentVLength := (9.14 / (2 * 14)) * h.hex.hexHeight
+	hexSegmentHLength := (4.8 / 7.5) * h.hex.hexWidth
+
+	h.segmentObjects[0].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+	h.segmentObjects[1].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+	h.segmentObjects[2].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+	h.segmentObjects[3].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+	h.segmentObjects[4].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+	h.segmentObjects[5].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+	h.segmentObjects[6].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+
+	pos := image.Pt(0, 0)
```

Comment:
Using fyne.Position would be better as you would not lose precision

Reply:
Done in 541512d. 

This code was ported from a different toolkit, hence why there are some non-idiomatic things.